### PR TITLE
Introducing the deprecated flag on the backend.

### PR DESCRIPTION
**Pull request by isoos**
#289.

- tool to get/set/clear the flag in the DataStore
- `search` skips adding it to the index when the flag is set
- `analyzer` skips analysis if the flag is set
- package tag and score is updated to handle the status

Disclaimer: I haven't tested it on staging yet.

### CODE REVIEW

File: app/bin/tools/deprecate.dart [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    }
+  });
+}
+
+Future _read(String packageName) async {
+  final Package p = (await dbService
+          .lookup([dbService.emptyKey.append(Package, id: packageName)]))
+      .single;
+  if (p == null) {
+    throw new Exception('Package $packageName does not exists.');
```
Comment (jakobr-google) [inline app/bin/tools/deprecate.dart]: s/exists/exist/

File: app/lib/search/updater.dart [hunk 2]
```diff
@@ -55,7 +55,8 @@ class BatchIndexUpdater implements TaskRunner {
         final int count = _snapshot.documents.length;
         _logger
             .info('Got $count packages from snapshot at ${_snapshot.updated}');
-        await packageIndex.addPackages(_snapshot.documents.values);
+        await packageIndex.addPackages(
+            _snapshot.documents.values.where((pd) => pd.isDeprecated != true));
```
Comment (jakobr-google) [inline app/lib/search/updater.dart]: *Maybe* add a little note here, so some helpful person doesn't accidentally change it to `!pd.isDeprecated`.

File: app/lib/frontend/templates.dart [hunk 3]
```diff
@@ -224,7 +227,10 @@ class TemplateService {
 
     final Map<String, dynamic> data = {
       'package': package,
+      'show_depreacted': analysisStatus == AnalysisStatus.deprecated,
```
Comment (jakobr-google) [inline app/lib/frontend/templates.dart]: s/depreacted/deprecated/ (here, and in the template)

**Comment by isoos** [comment]
I'll test this tomorrow on staging before merging.

**Pull request merged**